src/app/business/services/dynamic-form/dynamic-form.component.ts

```diff
@@ -209,10 +217,87 @@ export class DynamicFormComponent implements OnInit {
         this.requestBody.user_id = this.default_parameters['user_id'];
         this.requestBody.name = formObject.instanceName;
         this.requestBody.description = formObject.instanceDescription;
-        this.requestBody.parameters = _.omit(formObject, ['instanceName', 'instanceDescription']);
+        if(this.showAnalysis){
+          let analysisParam = this.checkAnalysisParameters(formObject);
+          this.requestBody.parameters = analysisParam;
+        }else{
+          this.requestBody.parameters = _.omit(formObject, ['instanceName', 'instanceDescription']);
+        }
         this.createInstance(this.requestBody);
       }
 
+      checkAnalysisParameters(formObject){
+        if(!this.createCluster.clusterForm.valid){
+          for(let i in this.createCluster.clusterForm.controls){
+              this.createCluster.clusterForm.controls[i].markAsTouched();
+          }
+          return;
+        }
+        let requestParams = this.getRequestParam(this.createCluster.clusterForm.value,formObject);
+        return requestParams;
+      }
+      getRequestParam(value,formObject){
+        let configureValue = this.createCluster.setConfigureForm.value;
+        let actionOnFailure;
+        if(configureValue['action'] == "Cancel and wait"){
+          actionOnFailure = "CANCEL_AND_WAIT";
+        }else if(configureValue['action'] == "Terminate cluster"){
+          actionOnFailure = "TERMINATE_CLUSTER";
+        }else{
+          actionOnFailure = configureValue['action'].toUpperCase();
+        }
+        let params = {
+          ip_addr: location.hostname,
+          port: location.port,
+          name: formObject.instanceName,
+          tenant_id: this.paramStor.CURRENT_TENANT().split("|")[1],
+          analysis_engine_type: formObject.analysis_engine_type,
+          dest_backend_name: formObject.dest_backend_name,
+          src_bucket_name: formObject.src_bucket_name,
+          dest_bucket_name: formObject.dest_bucket_name,
+          analysis_args:{
+            Name: value.clusterName,
+            AK: value.ak,
```

Comment:
AK/SK should be passed as a header and not a request body param.

Reply:
But this AK/SK is not AK/SK  used for authentication, it's AK/SK as it does backend creation, and parameters that require user input.

Reply:
Any authorization and authentication information that is passed as a parameter will be available in plain view. Please check with @himanshuvar for this.